src/collision/collision_system.cpp

```diff
@@ -714,6 +714,22 @@ CollisionSystem::is_free_of_movingstatics(const Rectf& rect, const CollisionObje
   return true;
 }
 
+bool
+CollisionSystem::is_free_of_specifically_movingstatics(const Rectf& rect, const CollisionObject* ignore_object) const
+{
+  using namespace collision;
+
+  for (const auto& object : m_objects) {
+    if (object == ignore_object) continue;
+    if (!object->is_valid()) continue;
+    if ((object->get_group() == COLGROUP_MOVING_STATIC)) {
+      if (intersects(rect, object->get_bbox())) return false;
```

Comment:
These could be merged to single `if`.

Reply:
Do you want me to do that for the other functions with redundant if statements, or just this new one?

Reply:
Only these that you added, changing all of them would be out of this PR scope.